code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -24,7 +24,10 @@ import picocolors from 'picocolors';
 import sirv from 'sirv';
 import { convertPathToPattern } from 'tinyglobby';
 import { dedent } from 'ts-dedent';
+import type { PluginOption } from 'vite';
 
+// ! Relative import to prebundle it without needing to depend on the Vite builder
+import { withoutVitePlugins } from '../../../../builders/builder-vite/src/utils/without-vite-plugins';
```

Comment:
`without-vite-plugins` uses a `type` import from `vite`. That means that the `experimental-addon-test` also depends on Vite; otherwise, the `PluginOption` cannot be resolved in package manager environments, which are strict about peer dependencies. 

Generally, I like to avoid relative imports like this because of the likelihood that the package that imports from another package utils or functions doesn't get the dependencies right. It can easily be overseen. 

If I were you, I would copy-paste the helper function and remove the `PluginOptions` reference. Otherwise, `experimental-addon-test` needs to mention vite as a peer dependency.

Reply:
But type imports are stripped from the `dist`, so it wouldn't show up in the package, so it's not necessary. Unless the imported type made it's way into a generated `.d.ts` file, but it doesn't because it's not part of the public API.

Reply:
But it's fair to be defensive and say _maybe_ in the future that module would import from dependencies that were in the original package but not the sibling package, without considering the sibling package.

Reply:
LGTM 
![image](https://github.com/user-attachments/assets/f7cb4f7f-635b-4ad7-aed8-cd7870fa9b8e)